### ISSUE TITLE
Month names wrong/duplicated

### ISSUE THREAD

**Issue by softwaresteph** [Gruntfuggly/vscode-journal-view#6]
https://i.imgur.com/wZzVYjz.png

"September" views as "October"
"November" views as "December"

When I hover over the names, it shows the appropriate month number (couldn't get a screenshot though)

Things I tried: 
* Clicking "refresh".
* Quit and reopen.
* Disable, reload, Enable, reload the extension
* View folders in Finder to make sure the month numbers were correct

**Comment by Gruntfuggly** [comment]
Please update to the latest version. I found the same problem earlier today. If it still happens after updating to the latest, please let me know.

It's something to do with how the month name is calculated due to it being October the 31st!

**Comment by softwaresteph** [comment]
It appears to still be happening with version 0.0.17

**Comment by Gruntfuggly** [comment]
Is it still happening today? The bug I fixed was due to it creating a date based on the current day and then setting the month. When the date was the 31st it adjusted the date into the next month.

Sounds like that may not be your issue though.

If it's still happening, would you be happy to put in a couple of lines to debug it?

**Comment by maciematt** [comment]
I'm also on 0.0.17, and same problem here - November is showing as December.

**Comment by Gruntfuggly** [comment]
Could you do me a favour - if you look in 
`.vscode/extensions/Gruntfuggly.vscode-journal-view-0.0.17` (or wherever your extensions are installed) and open `dataProvider.js`

There is a function at line 34 - can you add a couple of lines of debug as follows:
```
var getMonth = function( number )
{
    var date = new Date( 0 );
    console.log( "a:" + date.toISOString() );
    date.setMonth( parseInt( number ) - 1 );
    console.log( "b:" + date.toISOString() );
    return date.toLocaleString( vscode.env.language, { month: "long" } );
};
```

and the let me know what comes out on the console?

**Comment by softwaresteph** [comment]
Got it! 

```
[Extension Host] a:1970-01-01T00:00:00.000Z
[Extension Host] b:1969-07-31T23:00:00.000Z
[Extension Host] a:1970-01-01T00:00:00.000Z
[Extension Host] b:1969-08-31T23:00:00.000Z
[Extension Host] a:1970-01-01T00:00:00.000Z
[Extension Host] b:1969-10-01T23:00:00.000Z
[Extension Host] a:1970-01-01T00:00:00.000Z
[Extension Host] b:1969-10-31T23:00:00.000Z
[Extension Host] a:1970-01-01T00:00:00.000Z
[Extension Host] b:1969-12-02T00:00:00.000Z
```

**Comment by Gruntfuggly** [comment]
That's wierd - for each month I get:
```
 [Extension Host] a:1970-01-01T00:00:00.000Z
 [Extension Host] b:1970-01-01T00:00:00.000Z
 [Extension Host] test:January
 [Extension Host] a:1970-01-01T00:00:00.000Z
 [Extension Host] b:1970-02-01T00:00:00.000Z
 [Extension Host] test:February
 [Extension Host] a:1970-01-01T00:00:00.000Z
 [Extension Host] b:1970-03-01T00:00:00.000Z
 [Extension Host] test:March
 [Extension Host] a:1970-01-01T00:00:00.000Z
 [Extension Host] b:1970-04-01T00:00:00.000Z
 [Extension Host] test:April
 [Extension Host] a:1970-01-01T00:00:00.000Z
 [Extension Host] b:1970-05-01T00:00:00.000Z
 [Extension Host] test:May
 [Extension Host] a:1970-01-01T00:00:00.000Z
 [Extension Host] b:1970-06-01T00:00:00.000Z
 [Extension Host] test:June
 [Extension Host] a:1970-01-01T00:00:00.000Z
 [Extension Host] b:1970-07-01T00:00:00.000Z
 [Extension Host] test:July
 [Extension Host] a:1970-01-01T00:00:00.000Z
 [Extension Host] b:1970-08-01T00:00:00.000Z
 [Extension Host] test:August
 [Extension Host] a:1970-01-01T00:00:00.000Z
 [Extension Host] b:1970-09-01T00:00:00.000Z
 [Extension Host] test:September
 [Extension Host] a:1970-01-01T00:00:00.000Z
 [Extension Host] b:1970-10-01T00:00:00.000Z
 [Extension Host] test:October
 [Extension Host] a:1970-01-01T00:00:00.000Z
 [Extension Host] b:1970-11-01T00:00:00.000Z
 [Extension Host] test:November
 [Extension Host] a:1970-01-01T00:00:00.000Z
 [Extension Host] b:1970-12-01T00:00:00.000Z
 [Extension Host] test:December
```

Could you add a couple of more lines of debug please:

```
var getMonth = function( number )
{
    console.log( "number:" + number );
    console.log( "lang:" + vscode.env.language );
    var date = new Date( 0 );
    console.log( "a:" + date.toISOString() );
    date.setMonth( parseInt( number ) - 1 );
    console.log( "b:" + date.toISOString() );
    return date.toLocaleString( vscode.env.language, { month: "long" } );
}
```

**Comment by softwaresteph** [comment]
Haha, bizzare! 
```
[Extension Host] number:07
[Extension Host] lang:en
[Extension Host] a:1970-01-01T00:00:00.000Z
[Extension Host] b:1969-07-31T23:00:00.000Z
[Extension Host] number:08
[Extension Host] lang:en
[Extension Host] a:1970-01-01T00:00:00.000Z
[Extension Host] b:1969-08-31T23:00:00.000Z
[Extension Host] number:09
[Extension Host] lang:en
[Extension Host] a:1970-01-01T00:00:00.000Z
[Extension Host] b:1969-10-01T23:00:00.000Z
[Extension Host] number:10
[Extension Host] lang:en
[Extension Host] a:1970-01-01T00:00:00.000Z
[Extension Host] b:1969-10-31T23:00:00.000Z
[Extension Host] number:11
[Extension Host] lang:en
[Extension Host] a:1970-01-01T00:00:00.000Z
[Extension Host] b:1969-12-02T00:00:00.000Z
```

**Comment by Gruntfuggly** [comment]
Err. OK, that's wierd. Can you try something different:
```
var getMonth = function( number )
{
    var date = new Date( 1970, parseInt( number ) - 1, 1 );
    console.log( "a:" + date.toISOString() );
    return date.toLocaleString( vscode.env.language, { month: "long" } );
}
```

**Comment by softwaresteph** [comment]
It worked! 

https://i.imgur.com/xyZh9Gv.png

```
[Extension Host] a:1970-07-01T05:00:00.000Z
[Extension Host] a:1970-08-01T05:00:00.000Z
[Extension Host] a:1970-09-01T05:00:00.000Z
[Extension Host] a:1970-10-01T05:00:00.000Z
[Extension Host] a:1970-11-01T05:00:00.000Z
```

**Comment by Gruntfuggly** [comment]
OK - I'll update the extension. To be honest I don't understand why it behaves differently, but I won't worry too much as it looks like a simpler way to do it anyway! 
Thanks for your help. :smiley:

**Comment by maciematt** [comment]
After downloading the update two months ago or so the issue disappeared, but now it seems to be back (see attached picture).

<img width="1440" alt="screen shot 2019-01-20 at 1 27 23 pm" src="https://user-images.githubusercontent.com/3673043/51443481-688a5580-1cb7-11e9-8a7f-f0f2ce53264b.png">

**Comment by Gruntfuggly** [comment]
I merged a pull request which fixed issue #8. It just changed line 36 of `dataProvider.js`  from
```
var date = new Date( 1970, parseInt( number ) - 1, 1 );
```
to
```
var date = new Date( 1970, parseInt( number ) - 1, 15 );
```
Can you try setting it back and seeing if it fixes the issue?

**Comment by maciematt** [comment]
I didn't explicitly update the plugin, but it looks like it maybe auto-updated, as the problem is now gone. I just checked, and I'm on vscode-journal-view v. 0.0.22.
